sessionctx/variable/varsutil.go

```diff
@@ -156,6 +162,197 @@ func ValidateGetSystemVar(name string, isGlobal bool) error {
 	return nil
 }
 
+// ValidateSetSystemVar checks if system variable satisfies specific restriction.
+func ValidateSetSystemVar(vars *SessionVars, name string, value string) (string, error) {
+	if strings.EqualFold(value, "DEFAULT") {
```

Comment:
Maybe we need to consider `System`?  Currently, `set time_zone='System'` does not work.

Reply:
@zhexuany it's not a common check, I think we'd better by refine `parseTimeZone` in `varsutil.go` to :  
```go
func parseTimeZone(s string) (*time.Location, error) {
	if strings.EqualFold(s, "SYSTEM") {
	.......
}
```

Reply:
It seems that we should validate the result format in `ValidateSetSystemVar`. Only modify `parseTimeZone ` is not enough. I'll fix here.

Reply:
```sql
tidb> set time_zone='System';                                                                                                                                               Query OK, 0 rows affected (0.00 sec)

tidb> select @@time_zone;
+-------------+
| @@time_zone |
+-------------+
| System      |
+-------------+
1 row in set (0.00 sec)
```
In MySQL:

```sql
mysql> set time_zone='System';                                                                                                                                              Query OK, 0 rows affected (0.00 sec)

mysql> select @@time_zone;                                                                                                                                                  
+-------------+
| @@time_zone |
+-------------+
| SYSTEM      |
+-------------+
1 row in set (0.01 sec)


```

Reply:
@zhexuany  Done here. PTAL